pagefaultgames/pokerogue
# [Bug] Fixes Transform/Imposter not updating type/battle stat changes immediately; set move PP to 5 when transforming

## What are the changes?

Transform/Imposter now properly update type and battle stat changes on use/switch in, respectively.
Transform/Imposter now properly set move PP to 5 (if the move's PP is greater than 5) when transforming

## Why am I doing these changes?



It's a bug.
#2564 
Transform/Imposter do not currently set the move PP to 5, as stated in [Bulbapedia](https://bulbapedia.bulbagarden.net/wiki/Transform_(move)#:~:text=When%20a%20move%20is%20copied%2C%20it%20will%20have%205%20PP%2C%20unless%20its%20maximum%20PP%20is%20less%20(such%20as%20Sketch).) 

## What did change?


Added `user.updateInfo()` to the end of the `TransformAttr` and the `PostSummonTransformAbAttr` to refresh the user's changes.
Placed the copy code of `PostSummonTransformAbAttr` into a `Promise` to mirror `TransformAttr`. 
Transform/Imposter now set the move's pp to 5 using a negative ppUp value, if the move's pp is greater than 5.

### Screenshots/Videos


Before (Transform)

https://github.com/user-attachments/assets/ff9e0561-e45c-4115-aaeb-5229523a66db

After (Transform)

https://github.com/user-attachments/assets/2473d6e0-a9a4-461a-aa16-0e6110f48f06

Before (Imposter)

https://github.com/user-attachments/assets/d6c58364-acec-45ee-bded-d12734086f50

After (Imposter)

https://github.com/user-attachments/assets/aeba731c-0d3a-4e56-9440-87591183154e

## How to test the changes?




Use src/overrides.ts to give your pokemon Transform or Imposter, check stat changes, type changes, and PP value are updated correctly.

## Checklist
